src/Illuminate/Routing/Router.php

```diff
@@ -1199,6 +1199,40 @@ public function currentRouteNamed(...$patterns)
         return $this->current() && $this->current()->named(...$patterns);
     }
 
+    /**
+     * Alias for the "isWithParameters" method.
+     *
+     * @param  mixed  ...$patterns
+     * @return bool
+     */
+    public function isWith(...$patterns)
+    {
+        return $this->isWithParameters(...$patterns);
+    }
+
+    /**
+     * Determine if the current route matches a pattern (named routes, with or without parameters).
+     *
+     * @param  mixed  ...$patterns
+     * @return bool
+     */
+    public function isWithParameters(...$patterns)
+    {
+        foreach ($patterns as $pattern) {
+            if (is_string($pattern)) {
+                if ($this->currentRouteNamed([$pattern])) {
+                    return true;
+                }
+            } else {
+                if (url()->current() == route($pattern[0], $pattern[1] ?? [])) {
```

Comment:
Can't make call to foundation helper function here.

Reply:
Thanks, We need full URI of site, what is your suggestion to get that without foundation helper?

Reply:
Now using `$this->getCurrentRequest()->url()` rather than `url()->current()`